### DISCUSSION TITLE
New Sonoff TX V2.0 ? How to flash

### DISCUSSION THREAD

**Discussion by henkiejan1** [arendst/Tasmota#14184] [Q&A] [answered]
Hi,

Today i received my second ordered Sonoff TX T0 1 gang EU. But when i opened the devices i was a little bit confused. It looks like a new version of this switch. The flash pins are simple to find. But how about GPIO0? I add some pictures in this post.

As you can see there is also some silicon tape or something added between the switch and the plastic front. Maybe to protect the switch for ghost switching. Little detail: if you remove the front, the PCB with the ESP8266 is little bit sitcked to the front and falls off. ;)

![20211227_205455](https://user-images.githubusercontent.com/40475578/147503951-8ef2da2d-5742-41a2-a6aa-39e0c0704ee4.jpg)

![20211227_205605](https://user-images.githubusercontent.com/40475578/147503970-568704cf-07e8-4a43-8236-42bceb862798.jpg)

**Comment by sfromis** [comment, marked as answer]
Gpio0 is on the bottom row of pins on the ESP8285 chip, number 2 from the right. Image is not very clear, I can't point out good candidates for an easier location, but you could try following the trace on the PCB. When grounding gpio0, it should not hurt if you also ground the next pin to the right, which is gpio4. Makes for an easier resting place for a probe :-)

**Reply by henkiejan1** [reply]
Thank you for your reply. Maybe should i try a little bit more before i open a question. ;) I got it flashed! Simple like the V1.0 one. GND connected to TP0 (KEY 1). Hold it connected when flashing and that's it!

I got some better pictures but file transfer between my phone and computer has troubles. I upload better pictures when it's fixed. ;)

Thank you for your quick response! :)

**Reply by sfromis** [reply]
If you got it flashed, great!

**Reply by gfwilliams** [reply]
I just tried and there's no `TP0` on my unit, but there is `TP1 KEY0` - shorting that to GND and applying power while pressing the button works great

**Comment by henkiejan1** [comment]
Just some pictures with natural light.
![20211227_205516](https://user-images.githubusercontent.com/40475578/147604431-d899d3f0-c2cc-46f9-806a-ced4001e7500.jpg)
![20211227_205413](https://user-images.githubusercontent.com/40475578/147604447-b40a9b1e-7f24-4af5-9cb2-5f49f8f78244.jpg)
![20211227_205225](https://user-images.githubusercontent.com/40475578/147604467-908e7898-f55c-4079-b6d7-b1acae011176.jpg)

**Comment by henkiejan1** [comment]
Maybe uasefull to publish something about this V2.0 on this link? https://tasmota.github.io/docs/devices/Sonoff-T1/

Positive: i got no ghost switching!

**Comment by RoelVB** [comment]
In case someone is looking for this.
For the T0EU2C I had to connect TP1/KEY0 to GND to start in flashmode

**Reply by tayhall** [reply]
This is the easiest option.

**Comment by barto95100** [comment]
For information and picture this is the TP1/KEY0 on V2.0 model T0EU1C=

![IMG_1145](https://user-images.githubusercontent.com/17863254/201542789-d46a310b-ffff-4856-8f7d-acfa5a5eafc3.jpg)

**Reply by mauryc55** [reply]
How did you flash it? Mine seem to be going into flash mode but NodeMCU can't connect to it. Just connecting.....____.....____ and nothing. Old ones worked just fine

**Reply by sfromis** [reply]
Did you ground gpio0 while applying power? Did you wire correctly (straight, not crossed), and disable the ESP8266 on the NodeMCU?

More generally, a more reliable strategy than hunting around for resistors or test points for gpio0 is to ground gpio0 directly on the [ESP8285/ESP8266](https://tasmota.github.io/docs/Pinouts/#esp8266-and-esp8285) chip. Yeah, tiny pins, but you need not worry too much, as grounding either neighbor pin at the same time is no problem.

**Reply by mauryc55** [reply]
I've soldered everything like on the old ones and tried:
1. Power up with GND to TP1/KEY0 and letting it go after 3 sec.
2. Power up with GND soldered to TP1/KEY0 leaving it that way.

It behaves like in flash mode but no connection. Tryed 3 different units. With the old ones I just power them on with GND to R19 for 3 sec and everything flyes.

"...and disable the ESP8266 on the NodeMCU?" What do You mean? With my old ones I just set this up. 
<img width="685" alt="Zrzut ekranu 2022-12-7 o 12 26 58" src="https://user-images.githubusercontent.com/120015497/206167541-54b3e79f-fe7c-41df-8dbf-69ea04b605cf.png">

**Reply by mauryc55** [reply]
I will try to ground it directly to the chip when plugging in. Only for a sec or two, right? If my setup works for old ones it should work with te new also, right?

**Reply by sfromis** [reply]
Yes, a couple of seconds should do it. I usually linger a bit, in case of capacitors needing to be "charged". As the word "setup" is not explicitly defined, I have no answer to that.

What I mean by disable the ESP8266 is you indicating that you are using a NodeMCU (a board with a ESP8266, can also be used as USB/serial adapter). If you did not mean that, but used "NodeMCU" to describe something which is not a NodeMCU, ignore that.

Better use a [recommended tool](https://tasmota.github.io/docs/Getting-Started/#flashing-tool) for flashing, instead of some "random" stuff. (Not seeing an outright issue with what you posted, but better take the easier route).

**Reply by mauryc55** [reply]
I'm using this:
<img width="523" alt="Zrzut ekranu 2022-12-7 o 12 51 13" src="https://user-images.githubusercontent.com/120015497/206172610-10c72638-a7db-4c7f-81de-bf0d09166bf2.png">
<img width="551" alt="Zrzut ekranu 2022-12-7 o 12 51 37" src="https://user-images.githubusercontent.com/120015497/206172630-88718337-1a8c-4173-89d7-eb3e3ade6ffd.png">

and flashing (HAA, not Tasmota) with NodeMCU PyFlasher app on Mac.

**Reply by sfromis** [reply]
I suspected both. Not using a recommended flashing tool. And your USB/serial adapter is a known dubious one, not really able to supply the power for *reliable* flashing. While this does not mean that it can never work, it is a very clear risk factor (many users have had trouble with that one), but you still need to be very particular about your whole setup, which is all the details, not just what adapter you are using.

**Reply by mauryc55** [reply]
Thank YOU. I'm just learning that to menage things in my house.
<img width="729" alt="Zrzut ekranu 2022-12-7 o 13 03 46" src="https://user-images.githubusercontent.com/120015497/206175009-e4a13475-b462-4ccb-8ff0-b5f8a62f2fb2.png">
This one's good?

**Reply by sfromis** [reply]
Yup, that's the one we recommend. The small U2 near the jumper for 3.3V/5V is a LDO voltage regulator for 150mA of 3.3V power, which is much better. But still, you should triple-check exact details of wiring and booting in your current setup.

**Reply by mauryc55** [reply]
I wanted to thank you very much! After changing my USB/serial adapter to the one above everything works smoothly!

**Comment by henkiejan1** [comment]
I has used a CP2104 adapter from Robotdyn. 
![afbeelding](https://user-images.githubusercontent.com/40475578/206855321-79f2aebc-56bf-4fde-8ff4-5e028f24ad35.png)

BTW i see that in the intruction page there is just no reference to this V2.0 version? Maybe just an idea?

**Reply by sfromis** [reply]
There are lots of USB/serial adapters out there, any reason to specifically add this expensive adapter which does not seem to be particularly great?

Saw a suggestion that it by default uses 5V signals (unless board altered) which is a bad starting point for ESP chips where specs are for 3.3V signals.

**Reply by barbudor** [reply]
Lacking a 3.3V regulator too
So may be create an unrecommended category 😊

**Reply by sfromis** [reply]
Specs suggests the chip having an internal voltage regulator of 100 mA, some of which is used by the chip itself. While not as good as the recommended one, at least better than that red "FTDI" fake clone many seem to start with. Found some sales listings suggesting CP2104 providing 180mA of 3.3V power, which would be just fine, but my confidence in larger numbers than the data sheet is not all that great.

**Comment by ojaksch** [comment]
Just bought two of these TX T2EU which are looking like the same as in the 1st picture. To enter the flashing mode I have chosen the GPIO0 pin of the ESP8285, but instead of it's fiddly direct pin I used the capacitor next to the pin (C23?) which is pretty easy to connect to GND. No issue flashing both devices.  

![sonoff_TX_T2EU](https://github.com/arendst/Tasmota/assets/17717333/a3d0ce43-96af-456e-8e9f-7f3e8e9e0fe9)

**Reply by sfromis** [reply]
Yeah, as long as you already spent fiddly time working out where they've "hidden" an alternate point for IO0 on some particular device, that's great, instead of the generally faster solution of simply using two pins on the chip. You do not need to point exactly at only one pin, you can safely ground the next outer one at the same time. Not fiddly, as a pin even rests in a more stable way there than on some soldered resistor.

**Reply by ojaksch** [reply]
This is totally correct and I did this with the Sonoff RF bridges some years ago but now I'm years older and my eyes and fingers aren't _that_ young and tremor-free any more. It was easier for me to match the component near to the ESP instead :-)

**Reply by henkiejan1** [reply]
I should try. I ordered a T0 3 gang but not got the time to test. I used a time ago a wemos d1 mini as usb to serial and works complete on 3.3V

**Reply by martintuerk** [reply]
Hello all,
I do have a T3EU2C switch V2.0 here which I'm not able to convince to go into flash mode. A T0EU2C with the very same board works just fine. Whenever I power the board on while having IO0 hold to ground the device answeres with a single red flash followed by two red flashes ... no blue wifi led, but no flash mode either. 
What am I missing here ? Board is the same as on the foto send by ojaksch but with additional pieces on it, since its a T3 and a 2 gang switch. But nevertheless holding IO0 to ground should get the esp8285 into flash mode, which it does NOT!

Any ideas ?

**Reply by samodelka** [reply]
It is definitely due to the insufficient voltage of 3.3V. I had the same problem with the T1EU1C v2 board. The chip was only sporadically or not recognized at all. The flashing process stopped in the middle. The USB2TTL converter I used worked perfectly in the past with the other Sonoff products and other ESP82xx boards. The solution was to supply the board with a separate power supply of 3.3 volts via AMS1117.

**Reply by martintuerk** [reply]
Good hint, but I already tried that. Can't flash it using an USB2TTL neither does it get into flash mode when hooked up to a raspberry pi. In the latter case I also tried a lab power supply for providing the 3.3V. Have not used the lab power supply together wit the USB2TTL, though. Will try after the holydays.

**Reply by sfromis** [reply]
Exactly what are you doing to "hold IO0 to ground"? That step can be a bit tricky, and depending on the exact board you have. And you need to connect to ground before and while applying power.

And of course, triple-checking and verifying the wiring, along with available 3.3V power, is important.

**Reply by martintuerk** [reply]
I know, did this a couple of times. Before connecting 3.3V I connect the GPIO0 of the ESP Chip to 0V and hold it there while connecting the 3.3V. I've tried this either by connecting the respective 8285 Pin or the testpoint on the board to zero like discribed in detail above. The T3EU2C and the T0EU2C are exactly the same board, the only difference is the number of pieces on the board, the T3 has some additional 433MHz features. Anyhow, flashing the T0 works like a charm, Ive already flashed a couple of those, the T3 does not, regardless what else I try ...  I'm also pretty sure it's not broken, I can boot the standard sonoff firmware and the relays are working ...
But I agree, wiring is tricky, especially since I'm not using any soldering ... 

Has anybody done this with a T3 model ?

**Reply by skylarhays** [reply]
Any luck with this?

**Reply by samodelka** [reply]
> Any luck with this?

Today, I successfully installed Tasmota v15.2.0 on a Sonoff T3EU2C without any issues. Afterwards, I flashed it back to the stock firmware, added it to eWeLink, and performed a firmware update from version 3.8.0 to v3.8.2. Everything worked flawlessly. I tried flashing it back to see if the cloud functionality would still work, just as a fallback option :-)

**Reply by sebnolte** [reply]
I try to install Tasmota on a Sonoff T2EU2C.

I connect the cable as follow:
VCC = 3.3V
RX = TX
TX = RX
GND = GND

With Tasmotizer or esptool I'm able to flash the firmware an got the feedback it was successful. During the flash process I also get the feedback wich ESP it is and there is a WiFi module.

When I repower the ESP nothing happen. Only the LED4 on the board is on red. No AP-Mode or something else.
What else can go wrong, even when the flash process is successful?

**Reply by sfromis** [reply]
Typical issue is having the USB/serial adapter used for flashing also supply power to try to get on Wifi. Many of those can't handle that, you should instead power via mains.

**Reply by sebnolte** [reply]
I tried to use the normal power supply with the same result. After 10 minutes there is no tasmota wifi and the red LED is on.
The red LED is only off, when I connect the TP1 to GND like described here.
